### PR TITLE
Upselling extension: fetch relations for child products also

### PR DESCRIPTION
**Pull request by alexbridge**
# Pull Request Template

## Description

Upselling extension: fetch relations for child products also

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

### CODE REVIEW

File: frontend/components/connectors.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { connect } from 'react-redux';
+import { withRoute } from '@shopgate/engage/core';
+import { getBaseProductId } from '@shopgate/pwa-common-commerce/product/selectors/product';
+import { hasProductRelationsFiltered } from '../selectors';
+
+/**
+ * @param {string} type .
+ * @returns {Object}
+ */
+const makeMapStateToProps = type => (state, { route: { state: { productId } } }) => {
```
Comment (Carsten89) [inline frontend/components/connectors.js:10]: why take the productId from routeState? why not using withProduct/useProduct

File: frontend/subscriptions/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  subscribe(processProduct$, ({ action, dispatch }) => {
+    const { id: productId, baseProductId } = action.productData;
+
+    if (productPage.type) {
+      dispatch(fetchProductRelations({
+        productId,
+        type: productPage.type,
+      }));
+
+      if (baseProductId) {
```
Comment (Carsten89) [inline frontend/subscriptions/index.js:24]: what if productId and baseProductId are the same? will it not cause two request?

File: frontend/components/PDPSheet/index.jsx [hunk 3]
```diff
@@ -153,14 +152,7 @@ class PDPSheet extends Component {
...
- * @returns {Object}
- */
-const mapStateToProps = (state, props) => ({
-  productId: getBaseProductId(state, props),
-});
 
-export default connect(mapStateToProps)(PDPSheet);
+export default makeConnectProductWithRelations(productPageAddToCart.type)(
+  PDPSheet
+);
```
Comment (Carsten89) [inline frontend/components/PDPSheet/index.jsx]: new line missing. will fix it here and in other files.

File: frontend/subscriptions/index.js (hunk 2 above)
Comment (alexbridge) [reply frontend/subscriptions/index.js:24]: baseProductId is null in this case and if is evaluated to false

File: frontend/components/connectors.js (hunk 1 above)
Comment (alexbridge) [reply frontend/components/connectors.js:10]: Sheet is located outside of product context. We have no access to selection. Taking it from route works

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**